### PR TITLE
Fix building and producing configs when using installed dependencies

### PR DESCRIPTION
**Pull request by ewertons**
Please look at code for details.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -676,6 +678,7 @@ if(${use_installed_dependencies})
         FILES
             "configs/${PROJECT_NAME}Config.cmake"
             "configs/${PROJECT_NAME}Functions.cmake"
+           "configs/azure_iot_build_rules.cmake"
```
Comment (jspaith) [inline CMakeLists.txt]: >"configs/azure_iot_build_rules.cmake" [](start = 11, length = 37)

Nit - start of line does not align with others in same block.

Comment (ewertons) [reply CMakeLists.txt]: to be fixed.

Comment (ewertons) [reply CMakeLists.txt]: Done.

File: jenkins/linux_install_deps.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
```
Comment (jspaith) [inline jenkins/linux_install_deps.sh]: Please add this script to the .vsts-ci.yml for some platform(?).

Comment (ewertons) [reply jenkins/linux_install_deps.sh]: That will come afterwards. But it will happen

File: CMakeLists.txt [hunk 3]
```diff
@@ -580,6 +580,8 @@ if(NOT ${use_installed_dependencies})
     endif()
 endif()
 
+include(dependencies.cmake)
```
Comment (jebrando) [inline CMakeLists.txt]: Not this time but the next, we need to consolidate the umock-c either put all of it in dependencies.cmake or have it here.

**Pull request merged**